src/components/context/TimerContext.js

```diff
@@ -0,0 +1,269 @@
+  const [maxRound, setMaxRound] = useState(defaultSettings["rounds"]);
+  const [roundType, setRoundType] = useState("Work");
+  const [workLength, setWorkLength] = useState(defaultSettings["work"]);
+  const [restLength, setRestLength] = useState(defaultSettings["rest"]);
+  const [minuteSetting, setMinuteSetting] = useState(
+    defaultSettings["minutes"]
+  );
+  const [secondSetting, setSecondSetting] = useState(
+    defaultSettings["seconds"]
+  );
+  const [timerActive, setTimerActive] = useState(false);
+  const [btnActive, setBtnActive] = useState(false);
+  const [roundTime, setRoundTime] = useState(20);
+  const [resetFlag, setResetFlag] = useState(false);
+  const [fastForwardFlag, setFastForwardFlag] = useState(false);
+  const [congratsFlag, setCongratsFlag] = useState(false);
+  const [queuePos, setQueuePos] = useState(0);
+  const [queueActive, setQueueActive] = useState(false);
+  const [pauseFlag, setPauseFlag] = useState(false);
+
+  // Function to make getting the round time easier with less typing
+  const easyRoundTime = () => {
+    return calcRoundTime(
+      timerType,
+      minuteSetting,
+      secondSetting,
+      workLength,
+      restLength,
+      roundType
+    );
+  };
+
+  useEffect(() => {}, [secondSetting]);
+
+  //Sets percent and current time to settings
+  const setPTime = () => {
+    if (timerType === "Countdown" || timerType === "XY") {
+      setCurrentTime(easyRoundTime());
+      setPercent(100);
+    } else {
+      setPercent(0);
+      setCurrentTime(0);
+    }
+  };
+
+  //Grabs timer from queue and apply all settings
+  const getTimerFromQueue = (index) => {
+    const timer = getTimers(index, 1)[0];
+
+    if (timer !== undefined) {
+      setTimerType(timer.timerType);
+      setSecondSetting(timer.secondSetting);
+      setMinuteSetting(timer.minuteSetting);
+      setMaxRound(timer.roundSetting);
+      setWorkLength(timer.workLength);
+      setRestLength(timer.restLength);
+    }
+  };
+
+  //Trigger timer start/pause and calculate total time needed for percentage
+  const toggleTimer = () => {
+    if (!queueActive) {
+      setQueuePos(0);
+    } else {
+      setPauseFlag(!pauseFlag);
+    }
+    setTimerActive(!timerActive);
+    setQueueActive(true);
+    setBtnActive(true);
+    setCongratsFlag(false);
+  };
+
+  //Sets up users for the next round
+  const nextRound = (time, percent, type) => {
+    if (!!type) {
+      setRoundType(type);
+      //Only advance round if rest round is complete
+      if (roundType === "Rest") {
+        setCurrentRound(currentRound + 1);
+      }
+    } else {
+      setCurrentRound(currentRound + 1);
+    }
+    setRoundTime(easyRoundTime);
+    setPercent(percent);
+    setCurrentTime(time);
+  };
+
+  //Timer is finished so advance queue position
+  const timerFinished = () => {
+    setQueuePos(queuePos + 1);
+    setTimerActive(false);
+    setPTime();
+  };
+
+  //Force end of queue finish
+  const QueueFinished = (reset) => {
```

Comment:
```suggestion
  const queueFinished = (reset) => {
```
Generally speaking, capitalized means a react component. Use camel case for function names